tools/tsconfig-paths/index.ts

```diff
@@ -0,0 +1,285 @@
+/**
+ * Purpose of this script is to set correctly all required paths in `compilerOptions.paths` property in all our `tsconfig` files.
+ * Use after adding new library or new entry point, or moving libraries in file system.
+ *
+ * This script is based on number of assumptions:
+ * - libraries live in `core-libs`, `feature-libs`, `integration-libs` and `projects` directories (not in subdirectories!)
+ * - libraries have all dependencies specified in their `package.json`
+ * - all dependencies to other spartacus libs are peerDependencies
+ * - script is run from the root project directory `ts-node ./tools/tsconfig-paths/index.ts`
+ * - each entry point have it's own `ng-package.json` file
+ * - we have `prettier:fix` script for formatting files
+ * - libraries have `tsconfig.lib.json` files for configuration
+ * - libraries have `tsconfig.schematics.json` files for schematics configuration
+ * - all entry points are `.ts` files
+ */
+
+import { execSync } from 'child_process';
+import fs from 'fs';
+import glob from 'glob';
+import path from 'path';
+
+/**
+ * Paths to `package.json` files for all libraries.
+ */
+const librariesPaths = glob.sync(
+  '{core-libs,feature-libs,integration-libs,projects}/!(node_modules)/package.json',
+  {
+    ignore: [
+      'projects/storefrontapp-e2e-cypress/package.json',
+      'projects/dev-schematics/package.json',
+      'projects/storefrontstyles/package.json',
+      'projects/schematics/package.json', // excluded as it is treated differently than feature libraries
+    ],
+  }
+);
+
+// Utility functions
+function readJsonFile(path: string) {
+  return JSON.parse(fs.readFileSync(path, 'utf-8'));
+}
+
+function saveJsonFile(path: string, content: Object) {
+  fs.writeFileSync(path, JSON.stringify(content, undefined, 2));
+}
+
+function setCompilerOptionsPaths(tsconfigPath: string, paths: Object) {
+  const tsConfigContent = readJsonFile(tsconfigPath);
+  if (Object.keys(paths).length) {
+    tsConfigContent.compilerOptions.paths = paths;
+    saveJsonFile(tsconfigPath, tsConfigContent);
+  }
+}
+
+function joinPaths(...parts: string[]) {
+  return path.join(...parts).replace(/\\/g, '/');
+}
+
+function logUpdatedFile(path: string) {
+  console.log(`✅ Updated ${path}`);
+}
+
+console.log('\nAnalyzing project structure...');
+let libraries: {
```

Comment:
We can use const instead of let here, to be super-unnecessary-correct.
```suggestion
const libraries: {
```